art/core/MetricCalculator.py

```diff
@@ -28,8 +31,8 @@ class MetricCalculator:
 
     @classmethod
     def register_prepare(
-        cls,
-        prepare_func,
+        cls: Type[Self],
```

Comment:
Everything is nice for me maybe beside this one. `cls` is like a `self` when we decorate function with @classmethod. So I don't know if typing is necessary in this case

Reply:
SUre, I'd just stick to `Any`. Will change it